assets/src/components/parts/janus-mcq/MultipleChoiceQuestion.tsx

```diff
@@ -711,6 +715,7 @@ const MultipleChoiceQuestion: React.FC<PartComponentProps<McqModel>> = (props) =
 
   const handleItemSelection = (
     { value, textValue, checked }: ItemSelectionInput,
+    snapshot: any,
```

Comment:
I think that instead all that might be necessary is to make this into a callback and make sure it updates when `selectedChoices` does, because it should already have the updated value I think

Reply:
I have just made this function as useCallback. but it was not retaining the selectedChoices value. and there is another variable `multipleSelection ` which was being undefined after useCallback changes so i have added `model ` in the dependencies. 
Let me know what other changes need to be made.